### DISCUSSION TITLE
Integration of ym2413 chip Emulator

### DISCUSSION THREAD

**Discussion by pschatzmann** [pschatzmann/arduino-audio-tools#596] [Ideas]
Hello, my name is Holman Buenaventura, I hope you are well, I dared to write to you because I am a follower of your blog, I have used the stk library that I adapted for the esp32 board and it is an incredible job, I wanted to show you something that may interest you in In case you don't know the project https://github.com/digital-sound-antiques/emu2413, it is an emulator written in C of the ym2413 chip, around this emulator there are already a couple of projects one is the adaptation of the emulator to be used with the sony spresense board in the arduino ide, https://github.com/mizuhiki/Spresense2413, another is the emulator adaptation for the esp32 board using freeRTOS, I guess you are a very busy person, but I know from your blog that you are also someone who likes this topic, it would be great if someone with your talent and knowledge could adapt that emulator to be used in a similar way to the stk library, it seems to be a smaller code although I don't know what so demanding be even to the esp32 processor, but in case it can be adapted and optimized it would be another very good option to create homemade midi instruments that allow having several sounds.

**Comment by pschatzmann** [comment]
I created the https://github.com/pschatzmann/MD_YM2413 to provide the emulator and the corresponding API, so that it can be used together with the audio-tools.

The Emulator however does not sound great! I addition it uses so much RAM that you need to have a board with SPI RAM...

**Reply by habuenav** [reply]
wow, thank you very much for taking my comment into account, I was also pleasantly surprised by the speed with which I managed to do it. I'm going to try how it works.
